### DISCUSSION TITLE
Updates for CIS Win10 21H1

### DISCUSSION THREAD

**Discussion by StWightman** [0x6d69636b/windows_hardening#82] [General]
Hi, it looks like there have been numerous changes or updates to the CIS Win10 21H1 document that haven't been reflected in the csv yet. The Benchmark was updated February 15th of this year, a bunch of the IDs have changed, and I'm seeing some discrepancies between what the Benchmark states for recommendations and what the csv has. I would imagine other OS Benchmarks have also been updated, is there any mechanism to get the csv files updated? The stable build doesn't include Win10 21H1 files, so I think I'm looking at the newest ones from December last year.
Thank you for this, it looks like a great solution to creating a hardened environment. Cheers!

**Comment by 0x6d69636b** [comment]
Thanks for your feedback. As far as I can see CIS published a new benchmark for 21H2, but the naming of the document is misleading "_CIS Microsoft Windows 10 Enterprise Benchmark v1.12.0_". The 21H1 is still in version v1.11.0, so therefore the CSV is up-to-date. I update the CSVs when I have time for it. Everybody is welcome to contribute and create additional CSVs or make changes to the current ones. Sadly there is not automation process

**Reply by StWightman** [reply]
Thank you for the quick response. That Benchmark is what I'm comparing the CSV to, there's a note at the beginning of the document that while it's based on Enterprise it should be applicable to all Win10 versions. I saved the CSV as xlsx, once I've made all the updates I'll be glad to post it here for review. There are a number of items like "enabled:false" or a number for setting when the recommendation is multiple options that I still need to verify and review where I think there are conflicts so it will likely be early next week.

**Reply by 0x6d69636b** [reply]
The CIS Microsoft Windows Desktop List has usually the version in the name like _CIS Microsoft Windows 10 Enterprise Release 21H1 Benchmark v1.11.0_ but not for 21H2. That's why I missed it in the last update. However, I looked into it und created a new CSV version.

However, I would appreciate it if you would still make the list. Then there would be a good check that everything is correct in my case. You are better off using the Windows 11 list as the 21H1 as the basis CSV, which saves you effort. Or you can compare the new Windows 10 21H2 list right away.

**Reply by StWightman** [reply]
Okay, here's a spreadsheet with all of the corrections and changes I found. I didn't create a csv because I found changes to the 21H2 CSVs and wanted to have them vetted before that happened in case I messed things up. I tried to verify every enable/disable change and there are 3 sheets in the file.
1. The Win10 Enterprise 21H1 CSV, I was almost through it making updates compared to the 21H2 Benchmark
2. The Win10 Enterprise 21H2 CSV, with conditional formatting to show changes from the 21H1 sheet and comments on the changes I think need to be made. Please note that the registry path and registry item columns can show up as different just because the 21H1 sheet combined them (path:item) in many cases, and 21H2 sheet does not.
3. The 21H2 User sheet, I found a change and noted the items are out of numerical order

<file removed>

Please let me know if I can do anything more, and certainly point out any errors I have made.

Last minute thought - I wanted to verify that the HardeningKitty will work with the changed structure of registry path:item. You'll see in the file there are many of those that are different between the old and new files.

**Reply by 0x6d69636b** [reply]
Thank you very much for your work! I'll look into it, the user part should already be fixed.

What do you mean with the changed structure of the registry path?

**Reply by StWightman** [reply]
Sorry for the lack of clarity. In the 21H2 sheet a large amount of the two columns for registry path are yellow indicating they are different from the 21H1 sheet. That's because the 21H1 sheet put both in the registry path column as path:item instead of the two columns path and item. The 21H2 sheet does the latter. I did not verify each path was the same or correct in 21H2. I wasn't certain if that would make a difference, if HardeningKitty applies the finding list by making changes to the registry that it would be interpreted correctly. I don't like to assume anything and I'm new to your utility. Thanks!

**Reply by 0x6d69636b** [reply]
I started with a review of the Excel list, and the first comments are not correct:

- 18.3.1: The recommend value is still 0 (which means in this case enabled), see https://admx.help/?Category=security-compliance-toolkit&Policy=Microsoft.Policies.SecGuide::Pol_SecGuide_0201_LATFP
- 18.3.4: The recommend value is still 0 (which means in this case enabled), see https://admx.help/?Category=security-compliance-toolkit&Policy=Microsoft.Policies.SecGuide::Pol_SecGuide_0102_SEHOP
- 18.5.4.2: The recommend value is still 0 (which means in this case enabled), see https://admx.help/?Category=Windows_10_2016&Policy=Microsoft.Policies.DNSClient::Turn_Off_Multicast
- 18.5.11.2: The recommend value is still 0 (which means in this case enabled), see https://admx.help/?Category=Windows_10_2016&Policy=Microsoft.Policies.NetworkConnections::NC_AllowNetBridge_NLA

Could you please double-check the settings in your list?

**Reply by StWightman** [reply]
Excellent, thank you. I didn't know which of the many resources you included that would clarify if 0 and 1 were different for items. Where the Benchmark stated Enabled and the findinglist had 0 I tried checking the registry path and many of them didn't exist in my test Win 10 Enterprise system. Those four references do indeed clarify that for those 0 = Enabled. Now that I am aware of this specific resource, I can go back through the spreadsheet tomorrow and correct things tomorrow.

**Reply by 0x6d69636b** [reply]
Another way to find the correct finding is using the Group Policy Editor (gpedit.msc), configure the setting as described in the CIS Benchmark and check the registry value afterwards

**Reply by StWightman** [reply]
Here's the most recent, I've only included the 21H2 sheet with the updates. There were still a couple rows that my interpretation of admx.help confirmed the 0/1 change, two entries that seemed to be missing from the sheet, and at the very end there is one entry that I don't find in the CIS Benchmark yet is on admx.help. You can probably verify the status on that one. Please let me know if there's anything else I can do.
[CIS_21H2_compared_to_Benchmark_and_admx.help_20220323.xlsx](https://github.com/0x6d69636b/windows_hardening/files/8337000/CIS_21H2_compared_to_Benchmark_and_admx.help_20220323.xlsx)

**Reply by 0x6d69636b** [reply]
The settings 18.8.7.2 and 18.8.21.3 are correct in the list, I changed the recommenced value for 18.9.11.1.10. The two missing settings are there, they have new the IDs 18.9.14.x. I checked the ID's for ASR, they seem to match (see https://docs.microsoft.com/en-us/microsoft-365/security/defender-endpoint/attack-surface-reduction-rules-reference?view=o365-worldwide). The recommendation 18.9.108.4.2 is split in more than one registry. The ID 18.9.108.4.2.2 checks for the registry item BranchReadinessLevel (see https://admx.help/?Category=Windows_10_2016&Policy=Microsoft.Policies.WindowsUpdate::DeferFeatureUpdates)

**Reply by StWightman** [reply]
Great, thanks! I'm attaching a screenshot of the CIS_Microsoft_Windows_10_Enterprise_Benchmark_v1.12.0, which is where I found the new ASR ID, it looks like there's a conflict between those two sources. 
![ASR1](https://user-images.githubusercontent.com/15955635/159941042-f7eeb3b1-0aff-4cdf-87a5-14eb4cff8984.png)

Likewise with 18.9.108.4.2.2, I don't see the BranchReadinessLevel (recommended set to 16) in the Benchmark, it does split into 2 registry entries, to defer and to defer for X days. It certainly makes sense it should be included.
![CIS_PreviewBuilds](https://user-images.githubusercontent.com/15955635/159941178-a48acd21-f4ad-436a-929b-f4de44d53640.png)

I thought because the Benchmark is published by CIS that it would be the appropriate resource to follow for the CSVs, and I don't know their relationship with Microsoft and which bases their recommendations off of the other's.

**Reply by 0x6d69636b** [reply]
I double checked all ASR IDs, they are correct. The description was updated by Microsoft, but not the IDs. I'll double check 18.9.108.4.2.2 later

**Reply by 0x6d69636b** [reply]
Alright, CIS does not recommend any setting for the BranchReadinessLevel (anymore). However, if this policy is configured, the registry item will be set. Since CIS does not provide a recommendation, I'll change it to the pre-selected value (2) and a low severity

**Comment by StWightman** [comment]
BTW, I did some testing by creating a CSV from the 21H2 sheet, and when I run the command
- Invoke-HardeningKitty -Mode HailMary -Log -Report -FileFindingList .\lists\test_finding_list.csv
It works, the log shows a successful run, however the Report is blank with only column headers. Is this expected, or is it because the CSV is now different from the hardening.md and policyrules files? 

I've run a post-hardening report using the test CSV and have compared it to the backup CSV. I'm not able to attempt a restore from the backup though, it throws an error:
Cannot validate argument on parameter 'FileFindingList'. The "Test-Path $_" validation script
for the argument with value ".\lists\hardeningkitty_backup_win10b_finding_list_0x6d69636b_machine-20220322-110917.csv"
did not return a result of True. Determine why the validation script failed, and then try the command again.
At line:1 char:68

The files are in their original location with their original names.

Cheers!

**Reply by 0x6d69636b** [reply]
HardeningKitty does not work if you use a different structure for registry keys (path:item). The registry path and items must be separated because I did not implement a parsing for path:item. So if you use a different format, that would explain the blank result. And for the other issue, can you try the full path?

**Reply by StWightman** [reply]
Thanks. That was my confusion, The two CSVs are the CIS Windows 10 Enterprise 21H1 and 21H2 files from the HardeningKitty zip file. I didn't alter the registry path/item structure, so if I understand you correctly HardeningKitty will work with the 21H2 format. I will try with the full path in the morning.
Cheers!

**Reply by StWightman** [reply]
The full path worked, thanks for the suggestion. I'm starting fresh tomorrow, and will also look at the other CIS CSVs to see which ones have the combined path:item and which don't. It will be easy enough to pull the column into a new sheet and delimit at the colon and paste the results into the two columns, I'm just curious because that 21H1 CSV was included in your lists folder so shouldn't have worked properly. If I understand correctly.

**Reply by 0x6d69636b** [reply]
Sorry I don't get it, no list should have the combined path:item, because HardeningKitty can work with that. If there is one, this is a bug and must be fixed

**Reply by StWightman** [reply]
Thanks for the response. When I expanded to view all of the registry paths there are actually only a handful use the path:item format. I'm attaching screenshots. I think a lot of problems were generated from trying to update the 21h1 to 21h2 before you provided the 21h2 file, so I've removed that link above. Now that you created the 21h2 file, I sent just the 21h2 sheet and we've worked on this, that first file is obsolete anyway.
![21h1 path item](https://user-images.githubusercontent.com/15955635/159933602-ab61a3c0-df33-45d8-9ce5-6eaf41b9bdc6.png)
![21h2 path item](https://user-images.githubusercontent.com/15955635/159932339-a9083808-9185-4a67-9a51-bdc739c2e325.png)
.

**Reply by 0x6d69636b** [reply]
I checked the lists in the repo, there are (thankfully) no such combinations: :-) However, if you spot any, please create an issue or pull request
